src/theoretical.jl

```diff
@@ -64,6 +64,27 @@ Evaluate the variogram at points `x` and `y`.
 #------------------
 # IMPLEMENTATIONS
 #------------------
+"""
+    NuggetEfect(n)
+
+A pure nugget effect (random) variogram with nugget `n`
+"""
+@with_kw struct NuggetEffect{T,D} <: Variogram{T,D}
+  nugget::T = 0.0
+  sill::T = 0.0
+  range::T = 0.0
+  distance::D = Euclidean()
+end
+function NuggetEffect(n::T) where {T <: AbstractFloat}
+  @assert n > 0 "Nugget effect should be greater than zero"
+  NuggetEffect(nugget=n)
+end
```

Comment:
We don't need this outer constructor. Like the other variograms in this source file, we assume the user won't be creating variograms with negative nugget.

Reply:
The reason for this constructor is that `\gamma = NuggetEffect(nugget=0.2)` is seems redundant, whereas `\gamma = NuggetEffect(0.2)` is much better and cleaner. What do you think?

Reply:
So you are saying that `NuggetEffect(0.2)` doesn't work out of the box when you remove the sill and range fields?

Reply:
After removing those attributes, NuggetEffect(0.2) fails but NuggetEffect(0.2, Euclidean()) does not.

Reply:
In this case, we can just add a one-liner: `NuggetEffect(n) = NuggetEffect(nugget=n)` similar to what you had before.

Reply:
This is an example why I like Julia :-)